heuristic/repair_operators.py

```diff
@@ -16,44 +16,63 @@ def worst_week_repair(shifts_in_week, competencies, t_covered_by_shift, employee
         or we do not have any employees to assign to this shift as all employees are working this day 
     """
-    
+
     repair_set = []
     employees_changed = employees
     changed = destroy_set.copy()
     impossible_shifts = []
-
-    while(True):
+    
+    while True:
         calculate_deviation_from_demand(state, competencies, t_covered_by_shift, employee_with_competencies, demand, changed)
         delta_calculate_negative_deviation_from_contracted_hours(state, employees_changed, contracted_hours, weeks, time_periods_in_week, competencies, time_step)
-        shifts =    {
-                    (t1, v1, competencies): -sum(state.soft_vars["deviation_from_ideal_demand"][c,t] for c in competencies for t in t_covered_by_shift[t1, v1] if (c,t) in state.soft_vars["deviation_from_ideal_demand"])
-                                            - (20*(len(competencies)-1) + v1) for competencies in employee_with_competency_combination for t1, v1 in shifts_in_week[week[0]] if (t1,v1,competencies) not in impossible_shifts
-                    }
+
+        # TODO: Must discuss with Haakon that this works as needed
+        shifts = {
+                (t1, v1, comp):
+                sum(
+                    state.soft_vars["deviation_from_ideal_demand"].get((c, t), 0)
+                    for c in competencies
+                    for t in t_covered_by_shift[t1, v1]
+                )
+                - (20*(len(competencies)-1) + v1)
+                for comp in employee_with_competency_combination
+                for t1, v1 in shifts_in_week[week[0]]
+                if (t1, v1, comp) not in impossible_shifts
+                }
 
         shift = max(shifts.items(), key=itemgetter(1))[0]
 
-        deviation_from_demand = -sum(state.soft_vars["deviation_from_ideal_demand"][c,t] 
-                                    for c in shift[2]
-                                    for t in t_covered_by_shift[shift[0], shift[1]] 
-                                    if (c, t) in state.soft_vars["deviation_from_ideal_demand"] 
-                                    if state.soft_vars["deviation_from_ideal_demand"][c,t] < 0)
+        deviation_from_demand = -sum(state.soft_vars["deviation_from_ideal_demand"][c, t]
+                                for c in [shift[2]]
+                                for t in t_covered_by_shift[shift[0], shift[1]]
+                                if (c, t) in state.soft_vars["deviation_from_ideal_demand"]
+                                if state.soft_vars["deviation_from_ideal_demand"][c, t] < 0)
 
-        if(deviation_from_demand <= 6):
+        if deviation_from_demand <= 6:
             return repair_set
-        
-        
-        y_s_1 = {t: {(c): state.soft_vars["deviation_from_ideal_demand"][c,t] for c in shift[2] if (c,t) in state.soft_vars["deviation_from_ideal_demand"]} for t in t_covered_by_shift[shift[0], shift[1]]}
```

Comment:
If my understanding is correct, this line should be updated to: 
```python
y_s_1 = {t: {(c): state.soft_vars["deviation_from_ideal_demand"][c,t] 
    for c, t in shift[2] if (c,t) in state.soft_vars["deviation_from_ideal_demand"]} 
    for t in t_covered_by_shift[shift[0], shift[1]]}
```

Reply:
No. As far as I can see the line is correct. shift[2] is a tuple of competencies constituting a competency pair. e.g (0,1) is the competency-pair for competency 0 and 1. For each t in the shift we would like to know for which competency in the competency pair there is the most demand. We therefore loop over the competencies in the pair.